app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.java

```diff
@@ -0,0 +1,154 @@
+package net.squanchy.favorites.view;
+
+import android.content.Context;
+import android.support.annotation.IntDef;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+
+import java.util.Collections;
+import java.util.List;
+
+import net.squanchy.R;
+import net.squanchy.schedule.domain.view.Event;
+import net.squanchy.schedule.domain.view.Schedule;
+import net.squanchy.schedule.domain.view.SchedulePage;
+import net.squanchy.schedule.view.EventItemView;
+import net.squanchy.schedule.view.EventViewHolder;
+import net.squanchy.schedule.view.ScheduleViewPagerAdapter;
+import net.squanchy.search.view.HeaderViewHolder;
+import net.squanchy.support.lang.Lists;
+
+import org.joda.time.LocalDateTime;
+
+class FavoritesAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+
+    private final LayoutInflater layoutInflater;
+
+    @IntDef(value = {ItemViewType.TYPE_TALK, ItemViewType.TYPE_HEADER})
+    @interface ItemViewType {
+
+        int TYPE_TALK = 0;
+        int TYPE_HEADER = 1;
+    }
+
+    private Schedule schedule = Schedule.create(Collections.emptyList());
+
+    @Nullable
+    private ScheduleViewPagerAdapter.OnEventClickedListener listener;
+
+    FavoritesAdapter(Context context) {
+        setHasStableIds(true);
+        layoutInflater = LayoutInflater.from(context);
+    }
+
+    @Override
+    public long getItemId(int position) {
+        return findFor(
```

Comment:
Since we do two complex routines in the `getItemId` and `getItemViewType`, couldn't we just pre-parse the data, so that we just access the direct element when we need to?